test/factory_compatibility_table_test.rb

```diff
@@ -0,0 +1,230 @@
+# frozen_string_literal: true
+
+require_relative "test_helper"
+
+# Ensure that compatiblity table stays up to date. If this test fails
+# and your diff shows a checkmark instead of a red X, thanks for adding
+# that feature ! Otherwise, please take a look at your contribution.
+class FactoryCompatibilityTableTest < MiniTest::Test # :nodoc:
+  def test_table_ok
+    update_table_file(
+      generate_markdown(
+        compute_handled_methods_per_factory
+      )
+    )
+
+    assert(
+      system("git", "diff", "--quiet", "--", compatility_table_path),
+      proc {
+        require "pathname"
+        relative_path = Pathname.new(compatility_table_path)
+                                .relative_path_from(Pathname.new(Dir.pwd))
+        "Expected #{relative_path} not to change. Please check if this " \
+        "is intended and then commit changes, or fix otherwise."
+      }
+    )
+  end
+
+  private
+
+  def basic_method_handled?(geometry, method_sym, description)
+    if method_sym == :buffer
+      geometry.public_send(method_sym, 1)
+    else
+      geometry.public_send(method_sym)
+    end
+
+    true
+  rescue RGeo::Error::UnsupportedOperation
+    false
+  rescue NoMethodError => e
+    debug "NoMethodError", description, e.full_message(highlight: true)
+    false
+  rescue StandardError => e
+    debug "StandardError", description, e.full_message(highlight: true)
+    false
+  end
+
+  def basic_methods
+    @basic_methods ||= [
+      :factory,
+      :dimension,
+      :geometry_type,
+      :srid,
+      :envelope,
+      :as_text,
+      :as_binary,
+      :is_empty?,
+      :is_simple?,
+      :boundary,
+      :convex_hull,
+      :buffer
+    ].freeze
+  end
```

Comment:
We may also want to consider a way to add methods for specific geometry classes. For example, `Polygon#area`, `LineString#length`, etc.

We don't need to change this now, but we should make sure that these methods could be extended to account for those.

Reply:
@keithdoggett this will come later indeed !